test/mock.test.ts

```diff
@@ -31,7 +31,7 @@ for (const { type, noop, empty, proxy } of types) {
 
     describe("proxy", () => {
       it("nesting", () => {
-        const p = new proxy.foo.bar();
+        const p = new proxy.foo!.bar!();
```

Comment:
nothing important but the purpose of proxy is that they always do exist. is it something we can easily do? if not totally fine.

Reply:
I'm not aware of a way to tell TS that all possible properties are defined on an object. I think it requires explicit naming.